tests/integration.spec.coffee

```diff
@@ -1880,6 +1880,26 @@ describe 'SDK Integration Tests', ->
 					m.chai.expect(promise).to.eventually.have.property('pubnubPublishKey')
 					m.chai.expect(promise).to.eventually.have.property('listenPort')
 
+				it 'should be able to configure image parameters', ->
+					configOptions =
+						appUpdatePollInterval: 72
+						network: 'wifi'
+						wifiKey: 'foobar'
+						wifiSsid: 'foobarbaz'
+						ip: '1.2.3.4'
+						gateway: '5.6.7.8'
+						netmask: '9.10.11.12'
+						version: 'v1+foo'
+					promise = resin.models.os.getConfig(@application.id, configOptions)
+					# NOTE: the interval is converted to ms in the config object
+					m.chai.expect(promise).to.eventually.have.property('appUpdatePollInterval').that.equals(configOptions.appUpdatePollInterval * 60 * 1000)
+					m.chai.expect(promise).to.eventually.have.property('wifiKey').that.equals(configOptions.wifiKey)
+					m.chai.expect(promise).to.eventually.have.property('wifiSsid').that.equals(configOptions.wifiSsid)
+					m.chai.expect(promise).to.eventually.have.property('version').that.equals(configOptions.version)
+					m.chai.expect(promise).to.eventually.have.property('files')
+						.that.has.property('network/network.config')
+						.that.includes("#{configOptions.ip}/#{configOptions.netmask}/#{configOptions.gateway}")
```

Comment:
Each of these asserts returns a promise, but you only actually wait on the last promise here. If any except the last fails, the test will still pass (though you should get an 'unhandled rejection' error in the console).

These should be wrapped with a Promise.all.